boards/trex-one.json

```diff
@@ -0,0 +1,42 @@
+{
+  "build": {
+    "core": "arduino",
+    "cpu": "cortex-m0plus",
+    "extra_flags": "-D ARDUINO_RASPBERRY_PI_PICO -DARDUINO_ARCH_RP2040 -DARDUINO_TREX_ONE",
+    "f_cpu": "133000000L",
+    "hwids": [
+      [
+        "0x1209",
+        "0xD00F"
+      ]
+    ],
+    "mcu": "rp2040",
+    "variant": "TREX_ONE"
```

Comment:
It seems this board hasn't been added to the official Arduino framework, so you will need to use the original `RASPBERRY_PI_PICO` variant.

Reply:
That works for me, as long as the extra flag is set during build.